### ISSUE TITLE
Add --no-pub flag

### ISSUE THREAD

**Issue by rrousselGit** [dart-lang/sdk#45307]
It appears that new versions of `test` implicitly run `dart pub get` before executing tests.  
But as opposed to `flutter test`, `dart test` does not offer a `dart test --no-pub` flag.

This breaks some CI workflows, especially for mono-repositories relying on [melos](https//github.com/invertase/melos), which can no longer work because of this new behavior (which do not rely on `pub get` to resolve packages)

**Comment by rrousselGit** [comment]
It appears that `dart run` has the same issue

**Comment by natebosch** [comment]
https://github.com/dart-lang/pub/blob/0e657414a472e74ca5dd76ae0db50cc060251dec/lib/src/executable.dart#L294-L295

**Comment by natebosch** [comment]
This is coming from the pub code.

**Comment by mateusfccp** [comment]
This would help a lot.

For instance, I am working in a task for dart-sdk, and I can't run tests because running `dart test` will trigger `dart pub get`, which will override the dependencies and make the test fail. Running `dart run` won't trigger and will work, but this way I can't properly integrate tests with the IDE.

**Comment by sigurdm** [comment]
One problem here is that currently `dart test <args>` will pass all `<args>` to the testrunner. So there is no natural place to put --no-pub. For `dart run <executable> <args>` we can put it between  `run` and `<executable>`.

**Comment by rrousselGit** [comment]
To be fair, I'd also like some form of global config to disable it for an entire session.

Be it through an environment variable, like `PUB=false dart test` or some `dart config pub=false` or something.

**Comment by sigurdm** [comment]
Yeah, that would make more sense. Some way to point out that we don't want to use pub.
